components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/rabbitmq/RabbitMQUtils.java

```diff
@@ -0,0 +1,174 @@
+                handleException("Error while creating queue: " + queueName, e);
+            }
+        }
+    }
+
+    public static void declareQueue(Connection connection, String queueName,
+                                    Hashtable<String, String> properties) throws IOException {
+        Boolean queueAvailable = isQueueAvailable(connection, queueName);
+        Channel channel = connection.createChannel();
+
+        if (!queueAvailable) {
+            // Declare the named queue if it does not exists.
+            if (!channel.isOpen()) {
+                channel = connection.createChannel();
+                log.debug("Channel is not open. Creating a new channel.");
+            }
+            try {
+                channel.queueDeclare(queueName, isDurableQueue(properties),
+                        isExclusiveQueue(properties), isAutoDeleteQueue(properties), null);
+
+            } catch (IOException e) {
+                handleException("Error while creating queue: " + queueName, e);
+            }
+        }
+    }
+
+
+    public static void declareExchange(Connection connection, String exchangeName, Hashtable<String, String> properties) throws IOException {
+        Boolean exchangeAvailable = false;
+        Channel channel = connection.createChannel();
+        String exchangeType = properties
+                .get(RabbitMQConstants.EXCHANGE_TYPE);
+        String durable = properties.get(RabbitMQConstants.EXCHANGE_DURABLE);
+        try {
+            // check availability of the named exchange.
+            // The server will raise an IOException
+            // if the named exchange already exists.
+            channel.exchangeDeclarePassive(exchangeName);
+            exchangeAvailable = true;
+        } catch (IOException e) {
+            log.info("Exchange :" + exchangeName + " not found.Declaring exchange.");
+        }
+
+        if (!exchangeAvailable) {
+            // Declare the named exchange if it does not exists.
+            if (!channel.isOpen()) {
+                channel = connection.createChannel();
+                log.debug("Channel is not open. Creating a new channel.");
+            }
+            try {
+                if (exchangeType != null
+                        && !exchangeType.equals("")) {
+                    if (durable != null && !durable.equals("")) {
+                        channel.exchangeDeclare(exchangeName,
+                                exchangeType,
+                                Boolean.parseBoolean(durable));
+                    } else {
+                        channel.exchangeDeclare(exchangeName,
+                                exchangeType, true);
+                    }
+                } else {
+                    channel.exchangeDeclare(exchangeName, "direct", true);
+                }
+            } catch (IOException e) {
+                handleException("Error occurred while declaring exchange.", e);
+            }
+        }
+        channel.close();
```

Comment:
Closing **channel** isn’t strictly necessary here - it will be done automatically anyway when the underlying connection is closed.